Building with cppcheck fails to find `suppressions.txt`

*Description:*

Building with `ANALYZE_SOURCES_CPPCHECK` set to `ON` fails due to not finding `suppressions.txt`.

*To reproduce:*

In build dir, `ccmake .` set `ANALYZE_SOURCES_CPPCHECK` to `ON`, then `make`.

*Error message:*


```
hughcars@c889f3baddb0 build % cppcheck --version
Cppcheck 2.10
hughcars@c889f3baddb0 build % make -j6
[ 12%] Built target gklib
[ 25%] Built target metis
[ 37%] Built target parmetis
[ 50%] Built target superlu_dist
[ 62%] Built target hypre
[ 75%] Built target petsc
[ 87%] Built target gslib
[ 88%] Performing build step for 'palace'
-- ================= Configuring nlohmann/json dependency =================
-- Using the multi-header code from /Users/hughcars/AWS/palace/build/palace-build/_deps/json-src/include/
-- ====================== Configuring fmt dependency ======================
-- Module support is disabled.
-- Version: 9.1.0
-- Build type: Debug
-- CXX_STANDARD: 17
-- Required features: cxx_variadic_templates
-- ===================== Configuring MFEM dependency ======================
-- Building MFEM with zlib support for binary output compression
-- (optional) USER_CONFIG = /Users/hughcars/AWS/palace/build/palace-build/_deps/mfem-src/config/user.cmake
-- USE_XSDK_DEFAULTS = 'OFF'
-- MFEM: using package HYPRE
-- MFEM: using package SuperLUDist
-- MFEM: using package METIS
-- MFEM: using package GSLIB
-- MFEM: using package ZLIB
-- MFEM: using package MPI_CXX
-- MFEM build type: CMAKE_BUILD_TYPE = Debug
-- MFEM version: v4.5.1
-- MFEM git string: tags/v4.5-289-ga598682d68e5a21e84bc1c31205a4236e01515d2-dirty
-- TPL_INCLUDE_DIRS = /Users/hughcars/AWS/palace/build/include;/opt/homebrew/Cellar/open-mpi/4.1.4_2/include;/opt/homebrew/include;/Library/Developer/CommandLineTools/SDKs/MacOSX13.1.sdk/usr/include
-- Writing substitute header --> "mfem.hpp"
-- Writing substitute header --> "mfem-performance.hpp"
-- performance_ex1: add flags "-mcpu=apple-m1 -fcolor-diagnostics -fvectorize -fslp-vectorize -ffp-contract=fast"
-- performance_ex1p: add flags "-mcpu=apple-m1 -fcolor-diagnostics -fvectorize -fslp-vectorize -ffp-contract=fast"
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- CMAKE_INSTALL_PREFIX = /Users/hughcars/AWS/palace/build
-- ================== Finished dependency configuration ===================
-- Found PETSc: 3.18.99
-- PETSc test program - Failed
-- PETSc test program with static linkage - Success
-- Found SLEPc: 3.18.99
-- SLEPc test program - Success
-- Git string: v0.11.0-10-gb016c1b-dirty
-- Static analysis with clang-tidy not requested
-- Found cppcheck for static analysis: /opt/homebrew/bin/cppcheck
-- CMake build type: Debug
-- Building for architecture: arm64
-- Summary of extra compiler flags: 
-- Installation directory: /Users/hughcars/AWS/palace/build/bin
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/hughcars/AWS/palace/build/palace-build
[  0%] Built target fmt
```
long build sequence...
```
[ 69%] Linking CXX static library libmfem.a
[ 81%] Built target mfem
[ 81%] Building CXX object CMakeFiles/palace.dir/main.cpp.o
[ 81%] Building CXX object CMakeFiles/palace.dir/drivers/basesolver.cpp.o
[ 81%] Building CXX object CMakeFiles/palace.dir/drivers/eigensolver.cpp.o
[ 81%] Building CXX object CMakeFiles/palace.dir/drivers/magnetostaticsolver.cpp.o
[ 81%] Building CXX object CMakeFiles/palace.dir/drivers/electrostaticsolver.cpp.o
[ 84%] Building CXX object CMakeFiles/palace.dir/drivers/drivensolver.cpp.o
cppcheck: error: couldn't open the file: "/Users/hughcars/AWS/palace/build/palace-build/CMakeFiles/cppcheck/suppressions.txt".
make[5]: *** [CMakeFiles/palace.dir/main.cpp.o] Error 1
make[5]: *** Waiting for unfinished jobs....
cppcheck: error: couldn't open the file: "/Users/hughcars/AWS/palace/build/palace-build/CMakeFiles/cppcheck/suppressions.txt".
make[5]: *** [CMakeFiles/palace.dir/drivers/basesolver.cpp.o] Error 1
cppcheck: error: couldn't open the file: "/Users/hughcars/AWS/palace/build/palace-build/CMakeFiles/cppcheck/suppressions.txt".
make[5]: *** [CMakeFiles/palace.dir/drivers/eigensolver.cpp.o] Error 1
cppcheck: error: couldn't open the file: "/Users/hughcars/AWS/palace/build/palace-build/CMakeFiles/cppcheck/suppressions.txt".
make[5]: *** [CMakeFiles/palace.dir/drivers/magnetostaticsolver.cpp.o] Error 1
cppcheck: error: couldn't open the file: "/Users/hughcars/AWS/palace/build/palace-build/CMakeFiles/cppcheck/suppressions.txt".
make[5]: *** [CMakeFiles/palace.dir/drivers/electrostaticsolver.cpp.o] Error 1
cppcheck: error: couldn't open the file: "/Users/hughcars/AWS/palace/build/palace-build/CMakeFiles/cppcheck/suppressions.txt".
make[5]: *** [CMakeFiles/palace.dir/drivers/drivensolver.cpp.o] Error 1
make[4]: *** [CMakeFiles/palace.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [palace-cmake/src/palace-stamp/palace-build] Error 2
make[1]: *** [CMakeFiles/palace.dir/all] Error 2
```

*Environment:*
Clang, mac m1

--- a/cmake/StaticAnalysisHelpers.cmake
+++ b/cmake/StaticAnalysisHelpers.cmake
@@ -65,8 +65,9 @@ function(configure_cppcheck _target_name)
   )
   if(CPPCHECK_EXE)
     message(STATUS "Found cppcheck for static analysis: ${CPPCHECK_EXE}")
+    file(MAKE_DIRECTORY ${CMAKE_BINARY_DIR}/CMakeFiles/cppcheck)
     execute_process(
-      COMMAND     ${CMAKE_COMMAND} -E echo_append "*:${CMAKE_BINARY_DIR}/_deps/*"
+      COMMAND     ${CMAKE_COMMAND} -E echo "*:${CMAKE_BINARY_DIR}/_deps/*"
       OUTPUT_FILE ${CMAKE_BINARY_DIR}/CMakeFiles/cppcheck/suppressions.txt
     )
     set(CPPCHECK_COMMAND

--- a/cmake/StaticAnalysisHelpers.cmake
+++ b/cmake/StaticAnalysisHelpers.cmake
@@ -45,8 +45,13 @@ function(configure_clang_tidy _target_name)
         endif()
       endif()
       if(MPI_CXX_INCLUDE_DIRS)
+        set(CLANG_TIDY_EXTRA_ARG)
+        foreach(INCLUDE_DIR IN LISTS MPI_CXX_INCLUDE_DIRS)
+          set(CLANG_TIDY_EXTRA_ARG "${CLANG_TIDY_EXTRA_ARG} -I${INCLUDE_DIR}")
+        endforeach()
+        string(STRIP "${CLANG_TIDY_EXTRA_ARG}" CLANG_TIDY_EXTRA_ARG)
         list(APPEND CLANG_TIDY_COMMAND
-          "-extra-arg=-I${MPI_CXX_INCLUDE_DIRS}"
+          "-extra-arg=${CLANG_TIDY_EXTRA_ARG}"
         )
       endif()
     endif()

